### PR TITLE
Upgrade voting methods

### PR DESCRIPTION
**Pull request by tank157**
Second PR for (this Pivotal Card)[https://github.com/democracyworks/election-authority-works/pull/24]

In addition to changes to deal with the expanded `voting-methods`, I found recent changes to `voter-registration-methods` were causing the import process to fail, so I have some tweaks here to address that as well.

If you, dear reviewer, would like to try this all out, I suggest the following:

Add election-authority-works and voting-method-works to the docker-compose.yml in this project. Build it and run it. From election-authority-works, run an import like:

`lein run -m election-authority-works.import '{:host "localdocker" :port 5672}' ~/Downloads/EACopyofStates.csv` (where the EACopyofStates.csv is the exported "Copy of Sheets" from [here](https://docs.google.com/spreadsheets/d/1XxY1pkoiKNAM8nUjrDJ4MMm6x_rGWRHbreW9XfJtdjE/edit#gid=1736021287), and the port is adjusted accordingly

The from election-works, run
`lein run -m election-works.import '{:host "localdocker" :port 32779}' sample_election_import.csv` (again, adjust the port accordingly)

The contents of sample_election_import.csv is:
```
"date","description","type","registration-authority","vote-by-mail-authority","district-division","deadline:reg-by-mail","deadline:ballot-request","registration-deadline-type","election-day-registra
"2015-11-03","Test Election","Not Used","county","county","ocd-division/country:us/state:co/county:denver ocd-division/country:us/state:co/county:adams","2015-10-05","2015-10-23","postmarked","TRUE"
```

This should successfully import an election complete with merged in default voting methods.

### CODE REVIEW

**Comment by tank157** [comment]
Assigning to @danielglauser

File: dev-src/election_works/import.clj [hunk 1]
```diff
@@ -104,6 +130,18 @@
        (map (partial apply merge))
        set))
 
+(defn merge-voting-methods
+  "Merges left to right, overriding any prior voting-method with the
+  same settings for type and primary."
```
Comment (danielglauser) [inline dev-src/election_works/import.clj]: Works by extracting keys of interest with their values and building a new map with those maps as the keys, and the value being the original map. This allows us to match the overrides to the defaults only based upon the keys and values we are interested in.

File: resources/schemas/20160109000-add-voter-registration-method-options.edn [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ {:db/id #db/id[:db.part/db]
+  :db/ident :voter-registration-method.online/supports-iframe
+  :db/valueType :db.type/boolean
+  :db/cardinality :db.cardinality/one
+  :db/doc "Whether or not the online registration site can be iframed"
+  :db.install/_attribute :db.part/db}
+
+ ;; add form ref for :by-mail
+ {:db/id #db/id[:db.part/db]
+  :db/ident :voter-registration-method.by-mail/acceptable-form
```
Comment (danielglauser) [inline resources/schemas/20160109000-add-voter-registration-method-options.edn]: Should this be acceptable-forms?

Comment (tank157) [inline resources/schemas/20160109000-add-voter-registration-method-options.edn]: No, our standard is to still use singular names on cardinality many properties...for some reason I can't recall right now. But if you look around you'll see it to be true.

**Comment by danielglauser** [comment]
Well done sir. The additional tests really help. This was a lot of work. LGTM

**Pull request merged**